packages/binding-http/test/http-client-test.ts

```diff
@@ -456,4 +457,50 @@ class HttpClientTest2 {
                 .then(subscribeSpy);
         });
     }
+
+    @test "should unsubscribe successfully"(done: Mocha.Done) {
+        const client = new HttpClient();
+
+        // Subscribe to an event
+        const form: HttpForm = {
+            op: ["subscribeevent"],
+            href: `http://localhost:${port4}/`,
+        };
+
+        const app = express();
+        const server = http.createServer({}, app);
+
+        app.get("/", async (req, res) => {
+            res.send("Emitted Event!");
+        });
+
+        const eventSpy = chai.spy();
+
+        server.listen(port4, "0.0.0.0");
+        server.once("listening", async () => {
+            let counter = 0;
+            const sub = await client.subscribeResource(
+                form,
+                async () => {
+                    counter++;
+                    eventSpy();
+                    if (counter === 2) {
+                        sub.unsubscribe();
+                        // wait 100 ms, so that tests fail if unsubscribe didn't work
+                        await new Promise((resolve) => setTimeout(resolve, 100));
+                        // eslint-disable-next-line no-unused-expressions
+                        eventSpy.should.have.been.called.twice;
+                        server.close();
+                        done();
+                    }
```

Comment:
I think we should add a branch to stop this test so that it doesn't hang indefinitely if unsubscribing doesn't work..  maybe something like this

```

} else if (counter > 2) {
   server.close();
   done(new Error("unsubscribe didn't work as expected"));
}
```

Reply:
Done

Reply:
Great, thanks!